contracts/Aavegotchi/facets/AavegotchiLendingFacet.sol

```diff
@@ -0,0 +1,291 @@
+        require(rental.period == _period, "AavegotchiLending: Invalid rental period");
+        for (uint256 i; i < 3; i++) {
+            require(rental.revenueSplit[i] == _revenueSplit[i], "AavegotchiLending: Invalid revenue split");
+        }
+        address renter = LibMeta.msgSender();
+        address originalOwner = rental.originalOwner;
+        require(originalOwner != renter, "AavegotchiLending: renter can't be original owner");
+        require(rental.whitelistId == 0 || s.isWhitelisted[rental.whitelistId][renter], "AavegotchiLending: Not whitelisted address");
+
+        if (rental.amountPerDay > 0) {
+            uint256 transferAmount = rental.amountPerDay * rental.period;
+            require(IERC20(s.ghstContract).balanceOf(renter) >= transferAmount, "AavegotchiLending: not enough GHST");
+            LibERC20.transferFrom(s.ghstContract, renter, originalOwner, transferAmount);
+        }
+
+        rental.renter = renter;
+        rental.timeAgreed = block.timestamp;
+
+        uint256 tokenId = rental.erc721TokenId;
+        s.lentTokenIdIndexes[originalOwner][tokenId] = s.lentTokenIds[originalOwner].length;
+        s.lentTokenIds[originalOwner].push(tokenId);
+
+        if (rental.erc721TokenAddress == address(this)) {
+            LibAavegotchi.transfer(originalOwner, renter, tokenId);
+        } else {
+            // External contracts
+            IERC721(rental.erc721TokenAddress).safeTransferFrom(originalOwner, renter, tokenId);
+        }
+
+        // set original owner as pet operator
+        s.petOperators[renter][originalOwner] = true;
+
+        emit ERC721ExecutedRental(
+            _rentalId,
+            originalOwner,
+            renter,
+            rental.erc721TokenAddress,
+            tokenId,
+            rental.amountPerDay,
+            rental.period,
+            block.timestamp
+        );
+    }
+
+    ///@notice Allow a original owner to claim revenue from the rental
+    ///@dev Will throw if the NFT has not been lent or if the rental has been canceled already
+    ///@param _rentalId The identifier of the rental to claim
+    function claimAavegotchiRental(uint256 _rentalId) external {
+        AavegotchiRental storage rental = s.aavegotchiRentals[_rentalId];
+        require(rental.timeCreated != 0, "AavegotchiLending: rental not found");
+        require(rental.timeAgreed != 0, "AavegotchiLending: rental not agreed");
+        require(rental.completed == false, "AavegotchiLending: rental already completed");
+
+        address sender = LibMeta.msgSender();
+        address originalOwner = rental.originalOwner;
+        address renter = rental.renter;
+        address receiver = rental.receiver;
+        uint256 completionTime = rental.timeAgreed + rental.period * 1 days;
+        if (completionTime + 1 days > block.timestamp) {
+            require(originalOwner == sender, "AavegotchiLending: only owner can claim during agreement");
```

Comment:
The owner/lender of the Aavegotchi can repeated call the `claimAavegotchiRental` function to drain it of revenue tokens.
Example:   Aavegotchi has 100 tokens and the split for the owner is 10 percent.    The owner calls the `claimAavegotchiRental` function and gets 10 tokens.   Now the balance in the Aavegotchi is 90.   The owner then calls `claimAavegotchiRental` again and gets 10 percent of 90 which is 9,  and then the owner calls the function over and over again to keep getting 10 percent of the dwindling balance.

Reply:
Hm the way it's supposed to work is that the entire balance of each alchemica will be split when the owner claims. So if you had 100, the owner would receive 10 and the renter will receive 90. @orionstardust Can you check this please.

Reply:
@cinnabarhorse You are correct and it actually does work that way and is correct.   The problem I stated was incorrect.